src/renderer/components/widgets/channels/ChannelMessages.tsx

```diff
@@ -61,122 +46,117 @@ interface IChannelMessagesProps {
+  const scrollbarRef = React.useRef<Scrollbars>();
   const [offset, setOffset] = React.useState(0);
-  const updateSize = () => {
-    setOffset(0);
-  };
-  React.useEffect(() => {
-    window.addEventListener("resize", updateSize);
-  }, []);
+
   // TODO work on scroll behavior
   // React.useEffect(() => {
   //   setTimeout(() => {
   //     setLastScrollHeight(scrollbarRef.current.getScrollHeight())
   //   }, 0)
   // }, [contactId])
   // React.useEffect(() => {
-  //   console.log('tick')
-
   //   if (scrollbarRef.current) {
-  //     console.log(scrollbarRef.current)
   //     const currentHeight = scrollbarRef.current.getScrollHeight()
-  //     console.log(lastScrollHeight)
-  //     console.log(currentHeight)
-  //     console.log(`######################`)
-
   //     setLastScrollHeight(currentHeight)
   //     scrollbarRef.current.scrollTop(currentHeight - lastScrollHeight)
   //   }
   // }, [messages.size])
-  React.useEffect(() => {
-    if (msgRef.current && scrollbarRef.current) {
-      const margin =
-        msgRef.current.offsetHeight < scrollbarRef.current.getClientHeight()
-          ? scrollbarRef.current.getClientHeight() - msgRef.current.offsetHeight
-          : 0;
-      setOffset(margin);
-    }
-  }, [msgRef, scrollbarRef]);
-  // let username
-  // let tag
-  // if (isOffer) {
-  //   const msg = messages.toJS()[0]
-  //   tag = msg.message.tag
-  //   username = msg.sender.username
-  //   username = msg.sender.username
-  // }
+
+  const onScrollFrame = React.useCallback(
+    (e) => {
+      setScrollPosition(e.top);
+    },
+    [setScrollPosition]
+  );
+
   let groupedMessages: { [key: string]: DisplayableMessage[] };
   if (messages.length !== 0) {
-    groupedMessages = R.groupBy<DisplayableMessage>(
-      (msg) => {
-        return DateTime.fromFormat(
-          DateTime.fromSeconds(msg.createdAt).toFormat("cccc, LLL d"),
-          "cccc, LLL d"
-        )
-          .toSeconds()
-          .toString();
-      }
-    )(
+    groupedMessages = R.groupBy<DisplayableMessage>((msg) => {
+      return DateTime.fromFormat(
+        DateTime.fromSeconds(msg.createdAt).toFormat("cccc, LLL d"),
+        "cccc, LLL d"
+      )
+        .toSeconds()
+        .toString();
+    })(
       messages
         .filter((msg) => messagesTypesToDisplay.includes(msg.type))
         .concat(usersRegistration)
         .concat(publicChannelsRegistration)
         .sort((a, b) => a.createdAt - b.createdAt)
     );
   }
+
+  useEffect(() => {
+    if (
+      scrollbarRef.current &&
+      (scrollPosition === -1 || scrollPosition === 1)
+    ) {
+      scrollbarRef.current.scrollToBottom();
+    }
+    const eventListener = () => {
+      if (scrollbarRef.current) scrollbarRef.current.scrollToBottom();
+    };
+    window.addEventListener("resize", eventListener);
+    return () => window.removeEventListener("resize", eventListener);
+  }, [channelId, groupedMessages, scrollbarRef]);
+
+  React.useEffect(() => {
```

Comment:
I think this whole `useEffect` can be removed now. You are no longer using `margin` that this effect sets.
The positioning is now done by CSS, right?